Add title for most of dialogs from map's objects

ihhub opened the pull request:
- added titles and fixed messages for Pyramid dialogs ( close #4615, close #4613 )
- added titles for Skeleton dialogs ( close #4612 )
- added titles for Abandoned Mine dialogs ( close #4614 )
- added titles for Wagon dialogs

ihhub commented:
Hi @LeHerosInconnu, please take a look at these changes.

a1exsh commented on src/fheroes2/heroes/heroes_action.cpp:
```diff
@@ -1922,7 +1942,7 @@ void ActionToTreasureChest( Heroes & hero, const MP2::MapObjectType objectType,
                 StringReplace( msg, "%{art}", art.GetName() );
                 AGG::PlaySound( M82::TREASURE );
-                Dialog::ArtifactInfo( "", msg, art );
+                Dialog::ArtifactInfo( hdr, msg, art );
```
Hm, everywhere else this was named `title`...

ihhub replied:
The function had pre-defined variable called **hdr** so it was used here to fix missing header.

ihhub replied:
I understand that it's good to have a consistency in the code and make proper code but I haven't renamed the variable to minimize the number of changes in this file for now. It has a lot of problematic places.

oleg-derevenetz commented on src/fheroes2/heroes/heroes_action.cpp:
```diff
@@ -1350,48 +1363,47 @@ void ActionToPyramid( Heroes & hero, const MP2::MapObjectType objectType, s32 ds
-    }
-
-    if ( Dialog::YES == Dialog::Message( "", ask, Font::BIG, Dialog::YES | Dialog::NO ) ) {
+    if ( Dialog::YES == Dialog::Message( MP2::StringObject( objectType ), ask, Font::BIG, Dialog::YES | Dialog::NO ) ) {
```
Shouldn't we also use `title` here instead of calling `MP2::StringObject( objectType )` once again?

ihhub replied:
Ah, yes! I was reverting some changes in the code and forgot to add back.

oleg-derevenetz commented on src/fheroes2/heroes/heroes_action.cpp:
```diff
@@ -1537,6 +1551,8 @@ void ActionToPoorMoraleObject( Heroes & hero, const MP2::MapObjectType objectTyp
 
+    const std::string title( MP2::StringObject( objectType ) );
+
     if ( Dialog::YES == Dialog::Message( MP2::StringObject( objectType ), ask, Font::BIG, Dialog::YES | Dialog::NO ) ) {
```
Shouldn't we also use `title` here instead of calling `MP2::StringObject( objectType )` once again?

ihhub replied:
Yes, here as well. I was doing code refactoring but then decided not to expand this pull request.

oleg-derevenetz reviewed (changes requested):
Hi @ihhub I left a few minor notes here, please take a look when it's convenient for you.

LeHerosInconnu commented:
Hello @ihhub,

> Hi @LeHerosInconnu, please take a look at these changes.

- The yellow title "Abandoned Mine" is still missing for this window.
Also, the word "gold" has not been added if you wanted to do so in this PR.

Suggestion:
Also, I think the word "gold" should be added before the word "mine" in the text of the window after defeating the Ghosts.

"...restore the gold mine to production."

Instead of:

"...restore the mine to production."

![Yellow title check 01](https://user-images.githubusercontent.com/43583994/141679564-21fc8343-28c6-4ec1-8fbc-27ead8cbb29c.png)

- And in this window the word "the" was not added if you wanted to do it in this PR too.

Suggestion:
Also, I think the word "the" should be added before the name of the artifact.

"...you find the Golden Horseshoe."

Instead of:

"...you find Golden Horseshoe."

![Yellow title check 02](https://user-images.githubusercontent.com/43583994/141679576-262f0987-9f92-42ca-8ef4-a7356621e980.png)

Note that the word "the" is present in the text of the artifact in the Wagon window when the hero finds an artifact there.

![Yellow title check 03](https://user-images.githubusercontent.com/43583994/141680921-a72969e9-d782-4d8b-9b41-09b15ba4ea84.png)

For the other windows everything is correct.

ihhub commented:
Hi @oleg-derevenetz , I addressed your comments.

Hi @LeHerosInconnu , I added `the` word but haven't changed into gold mine to keep it as in the original game for now.
